src/StackYaml/Transformations/UpdateExtraDeps.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+module StackYaml.Transformations.UpdateExtraDeps
+    ( transformation
+    ) where
+
+import Control.Applicative         (many)
+import Control.Exception           (IOException, catch)
+import Control.Lens
+import Data.Aeson.Lens
+import Data.Char                   (isDigit)
+import Data.Function               ((&))
+import Data.List                   (intersperse)
+import Data.Monoid                 ((<>))
+import Data.Text                   (Text, pack)
+import Data.Yaml                   (Value)
+import System.Exit                 (exitFailure)
+import System.IO                   (hPutStrLn, stderr)
+import Text.Regex.Applicative.Text (anySym, few, match, psym, sym)
+
+import           Distribution.Package  (PackageIdentifier (..),
+                                        PackageName (..))
+import qualified Distribution.PackDeps as PackDeps
+import           Distribution.Version  (Version (..))
+
+transformation :: Value -> IO Value
+transformation value = do
+    newest <- catch PackDeps.loadNewest
+                    (\e -> hPutStrLn stderr (failedToLoad e) *> exitFailure)
+    traverseOf
+      (key "extra-deps" . values . _String)
+      (updateExtraDep newest)
+      value
+  where
+    failedToLoad :: IOException -> String
+    failedToLoad err =
+      unlines
+        [ show err
+        , "Failed to read package database. Doing 'cabal update' might help."
+        ]
+
+    updateExtraDep :: PackDeps.Newest -> Text -> IO Text
+    updateExtraDep newest dep =
+        (pkgName <$> parsePackageIdentifier dep)
+           & maybe
+               (dep <$ hPutStrLn
+                         stderr
+                         ("Ignoring invalid extra-deps entry " <> show dep))
+               (\name ->
+                   newestPackageId newest name
+                     & maybe
+                         (dep <$ hPutStrLn
+                                   stderr
+                                   ("Ignoring extra-deps entry "
+                                       <> show dep
+                                       <> " as it refers to unknown package "
+                                       <> show (unPackageName name)))
+                         (pure . pPrintPackageIdentifier))
+
+parsePackageIdentifier :: Text -> Maybe PackageIdentifier
+parsePackageIdentifier = match pkgIdentifierRe
+  where
+    pkgIdentifierRe =
+      PackageIdentifier
+        <$> (PackageName <$> few anySym <* sym '-')
+        <*> pkgVersionRe
+
+    pkgVersionRe = flip Version [] <$> ((:) <$> num <*> many (sym '.' *> num))
+      where
+        num = read <$> many (psym isDigit)
+
+pPrintPackageIdentifier :: PackageIdentifier -> Text
+pPrintPackageIdentifier PackageIdentifier{..} =
+    pack (unPackageName pkgName) <> "-" <> pPrintVersion pkgVersion
+
+pPrintVersion :: Version -> Text
```

Comment:
there is `showVersion` in `Data.Version` (probably re-exported from `Distribution.Version`